examples/typed-state/README.md

```diff
@@ -0,0 +1,263 @@
+# Typed State
+
+This example goes over how to use the typed state features in Burr with pydantic.
+
+It will cover the following concepts:
+
+1. Why you might want to use typing for your state in the first place
+1. Define typing at the application level
+1. Defining/altering state at the action level
+1. Doing so in a streaming manner
+1. Wiring that through to a FastAPI app
+
+This README will contain snippets + link out to the code.
+
+This adapts the [instructor + youtube example](../youtube-to-social-media-post/). This
+takes in a youtube video, creates a transcript, and uses OpenAI to generate a social media post based on that transcript.
+
+## Why type state?
+
+Burr originally came without typing -- typing is an additional (optional) plugin that requires some complexity. So, why use typing at all? And why use Pydantic? Lots of good reasons:
+
+### Typing provides guard-rails
+
+Typing state ensures you have guarentees about what the data in your state is, and what shape it takes. This makes it easier to work with/manipulate.
+
+### Typing makes development easier
+
+IDEs have integrations for handling type annotations, enabling you to
+avoid the cognitive burden of tracking types in your head. You get auto-completion on typed classes, and errors if you attempt to access or assign to a field that does not exist.
+
+### Typing makes downstream integration easier
+
+Multiple tools use types (especially with pydantic) to make interacting with data easier. In this example we use [instructor](https://python.useinstructor.com/blog/), as well as [FastAPI](https://fastapi.tiangolo.com/) to leverage pydantic models that Burr also uses.
+
+### Typing provides a form of documentation
+
+Type-annotation in python allows you to read your code and get some sense of what it is actually doing. This can be a warm introduction to python for those who came from the world of java initially (the authors of this library included), and make reasoning about a complex codebase simpler.
+
+## Setting up your IDE
+
+VSCode (or an editor with a similar interface) is generally optimal for this. It has
+pluggable typing (e.g. pylance), which handles generics cleanly. Unfortunately pycharm
+is often behind on typing support. See issues like [this](https://youtrack.jetbrains.com/issue/PY-44364) and [this](https://youtrack.jetbrains.com/issue/PY-27627/Support-explicitly-parametrized-generic-class-instantiation-syntax).
+
+While it will still work in pycharm, you will not get some of the beter auto-completion capabilities.
```

Comment:
```suggestion
While it will still work in pycharm, you will not get some of the beter auto-completion capabilities until those issues are resolved.
```